### ISSUE TITLE
Wrong posts displayed in timeline

### ISSUE THREAD

**Issue by Joshix-1** [tuskyapp/Tusky#4588]
### Detailed description

Sometimes shortly after loading more posts into the timeline near the top of it, the posts that are shown are wrong. They were near the top before but should've been replaced by new ones and when I click on them I get to different posts. This happened multiple times.

https://github.com/user-attachments/assets/5512480b-fc34-47a9-9f01-78a40d3090e0

### Steps to reproduce the problem

1. Look through timeline
2. Load more posts by swiping down on top
3. Sometimes when I click on a post it links to a different than the one displayed
...

### Debug information

```
Tusky 26.0

Device:

Gigaset E940-2878-03
Android version: 13
SDK version: 33

Account:

@joshix@fosspri.de
Version: 4.2.10+fosspride
```

Reading order: Newest first

**Comment by connyduck** [comment]
Are you logged in with multiple accounts, and if yes could this be the same bug where Tusky confuses posts of different accounts?

**Comment by Joshix-1** [comment]
i'm logged in with multiple accounts (mine & the bot account on my instance), but the other account doesn't follow the accounts in the video

**Comment by Joshix-1** [comment]
I think it happens mostly when I have a bad internet connection, but I'm not sure

**Comment by connyduck** [comment]
Is this the same post, boosted by the same person, but with different timestamps in the timeline twice? Weird, but that might be a clie what is happening.

Do you remember what you did before the video starts?

**Comment by Joshix-1** [comment]
> Is this the same post, boosted by the same person, but with different timestamps in the timeline twice? 
yes, but clicking on the top one leads to another post with the timestamp of one minute ago

> Do you remember what you did before the video starts?
not really, i probably just refreshed the timeline to load more posts
